generators/commons/src/project/AbstractProject.ts

```diff
@@ -0,0 +1,27 @@
+import { AbstractGeneratorContext } from "../AbstractGeneratorContext";
+import { AbsoluteFilePath } from "@fern-api/fs-utils";
+import { File } from "./File";
+
+export abstract class AbstractProject<GeneratorContext extends AbstractGeneratorContext> {
+    public readonly absolutePathToOutputDirectory: AbsoluteFilePath;
+    public readonly rawFiles: File[] = [];
+
+    public constructor(public readonly context: GeneratorContext) {
+        this.absolutePathToOutputDirectory = AbsoluteFilePath.of(this.context.config.output.path);
+    }
+
+    public addRawFiles(file: File): void {
+        this.rawFiles.push(file);
+    }
+
+    public async writeRawFiles(): Promise<void> {
+        for (const file of this.rawFiles) {
+            await file.write(this.absolutePathToOutputDirectory);
+        }
```

Comment:
can we make this ```
await Promise.all(this.rawFiles.map((rawFile) => await file.write(this.absolutePathToOutputDirectory)); 
```

WIll make sure we are writing files in parallel as fast as we can

Reply:
Good call, will update!